examples/vulkan/vk_hotload.rs

```diff
@@ -0,0 +1,246 @@
+    let shade_watcher = shade_runner::Watch::create(vert_path, frag_path).expect("failed to create watcher");
+    let shade_msg = shade_watcher
+        .rx
+        .recv()
+        .expect("Failed to receive shader")
+        .expect("failed to compile shader");
+    let vs = unsafe {
+        vk::pipeline::shader::ShaderModule::from_words(device.clone(), &shade_msg.shaders.vertex)
+    }
+    .unwrap();
+
+    let fs = unsafe {
+        vk::pipeline::shader::ShaderModule::from_words(device.clone(), &shade_msg.shaders.fragment)
+    }
+    .unwrap();
+
+    // The next step is to create a *render pass*, which is an object that describes where the
+    // output of the graphics pipeline will go. It describes the layout of the images
+    // where the colors, depth and/or stencil information will be written.
+    let render_pass = Arc::new(
+        vk::single_pass_renderpass!(
+            device.clone(),
+            attachments: {
+                // `color` is a custom name we give to the first and only attachment.
+                color: {
+                    // `load: Clear` means that we ask the GPU to clear the content of this
+                    // attachment at the start of the drawing.
+                    load: Clear,
+                    // `store: Store` means that we ask the GPU to store the output of the draw
+                    // in the actual image. We could also ask it to discard the result.
+                    store: Store,
+                    // `format: <ty>` indicates the type of the format of the image. This has to
+                    // be one of the types of the `vulkano::format` module (or alternatively one
+                    // of your structs that implements the `FormatDesc` trait). Here we use the
+                    // same format as the swapchain.
+                    format: app.main_window().swapchain().format(),
+                    // TODO:
+                    samples: app.main_window().msaa_samples(),
+                }
+            },
+            pass: {
+                // We use the attachment named `color` as the one and only color attachment.
+                color: [color],
+                // No depth-stencil attachment is indicated with empty brackets.
+                depth_stencil: {}
+            }
+        )
+        .unwrap(),
+    );
+
+    let pipeline = None;
+    // The render pass we created above only describes the layout of our framebuffer. Before we
+    // can draw we also need to create the actual framebuffer.
+    let view_fbo = RefCell::new(ViewFbo::default());
+
+    let mut model = Model {
+        render_pass,
+        pipeline,
+        vertex_buffer,
+        view_fbo,
+        shade_watcher,
+        shade_msg,
+        vert_shader: vs,
+        frag_shader: fs,
+        device,
+    };
+    update_pipeline(&mut model);
+    model
+}
+
+fn update(_app: &App, model: &mut Model, _: Update) {
+    let shader_msg = model.shade_watcher.rx.try_recv();
```

Comment:
We only really care about the last `shader_msg` that was received, so maybe something like this would be better:
```rust
let shader_msg = model.shade_watcher.rx.try_iter().last();
```
This way we skip any backed up intermediary messages (e.g. in the case where multiple file system events occurred at once for some reason).